kernel/src/process.rs

```diff
@@ -272,6 +273,15 @@ pub enum StoppedExecutingReason {
     KernelPreemption,
 }
 
+#[derive(PartialEq, Eq, PartialOrd, Ord)]
+pub struct BinaryVersion(NonZeroU32);
```

Comment:
I believe as it stands there is no way to read the actual version, only compare it with other versions. Perhaps just make the inner field `pub`?

```suggestion
pub struct BinaryVersion(pub NonZeroU32);
```
```

Reply:
I think there is no need to access the inner value directly. The Tock kernel only needs to compare two different binary versions. For debug purposes, I derived the `Debug` trait.

Reply:
OK, fine then to leave as-is. If we ever need to access it, we can add it.